internal/controller/apiextensions/claim/configurator.go

```diff
@@ -209,58 +227,69 @@ func (c *APIClaimConfigurator) Configure(ctx context.Context, cm resource.Compos
 		return nil
 	}
 
-	if err := merge(ucm.Object["status"], ucp.Object["status"],
+	udesiredCm, ok := cmPatch.(*claim.Unstructured)
+	if !ok {
+		return nil
```

Comment:
what does this mean? Similar to above: don't cast. Use the proper types directly.